lib/migrate.js

```diff
@@ -294,10 +304,15 @@ function saveExecutedMigrationsMetadata (migrations, options) {
     .reduce((chain, migration) => chain.then(() => r.table(migrationsTable).insert(migration).run(conn)), Promise.resolve())
 }
 
-function clearMigrationsTable (options) {
+function clearMigrationsTable (migrations, options) {
   const { r, conn, migrationsTable } = options
 
-  return r.table(migrationsTable).delete().run(conn)
+  const migrationsToDelete = []
+  migrations.forEach((item) => {
```

Comment:
I would only replace this `forEach` by a map. Something along the lines:
```javascript
return Promise.all(
  migrations.map(
    item => r.table(migrationsTable)
             .filter({filename: item.filename})
             .delete()
             .run(conn)
  )
)
```

@lirantal What do you think? Could you make this change and test? This would maintain the sort-of functional approach in the code.

Reply:
yep, sounds good and cleaner.